pkg/model/pool.go

```diff
@@ -53,12 +53,27 @@ type StoragePoolSpec struct {
 	// Default unit of FreeCapacity is GB.
 	FreeCapacity int64 `json:"freeCapacity,omitempty"`
 
-	// The storage type of the dock.
+	// The storage type of the storage pool.
 	// One of: "block", "file" or "object".
 	StorageType string `json:"storageType,omitempty"`
 
-	// Map of keys and json object that represents the extra epecs
+	// Map of keys and StoragePoolExtraSpec object that represents the properties
 	// of the pool, such as supported capabilities.
 	// +optional
-	Extras ExtraSpec `json:"extras,omitempty"`
+	Extras StoragePoolExtraSpec `json:"extras,omitempty"`
+}
+
+type StoragePoolExtraSpec struct {
+	// DataStorage represents suggested some data storage capabilities.
+	DataStorage DataStorageLoS `json:"dataStorage,omitempty" yaml:"dataStorage,omitempty"`
+	// IOConnectivity represents some suggested IO connectivity capabilities.
+	IOConnectivity IOConnectivityLoS `json:"ioConnectivity,omitempty" yaml:"ioConnectivity,omitempty"`
+	// DataProtection represents some suggested data protection capabilities.
+	DataProtection DataProtectionLos `json:"dataProtection,omitempty" yaml:"dataProtection,omitempty"`
+
+	// Besides those basic suggested pool properties above, vendors can configure
+	// some advanced features (diskType, IOPS, throughout, latency, etc)
+	// themselves, all these properties can be exposed to controller scheduler
+	// and filtered by selector in a extensible way.
+	Advanced map[string]interface{} `json:"advanced,omitempty" yaml:"advanced,omitempty"`
```

Comment:
How about rename "Advanced" to "ExtraProperties" or something else?

Reply:
We already defined `StoragePoolExtraSpec`, it may result in some confusions if we add another `ExtraProperties`.

Reply:
"Advanced" seems to say IOPS and BWS are not advanced because they are not under this category.  Anyway we can figure out a better name later.